pkg/chunk/aws/storage_client.go

```diff
@@ -620,6 +627,12 @@ func (a storageClient) getDynamoDBChunks(ctx context.Context, chunks []chunk.Chu
 				unprocessed.TakeReqs(requests, -1)
 				backoff.Wait()
 				continue
+			} else if ok && awsErr.Code() == validationException {
+				// this read will never work, so the only option is to drop the offending request and continue.
+				// TODO: add more debug options for capturing data/telemetry about the offending items?
+				level.Warn(util.Logger).Log("Error while fetching data from Dynamo: %v", awsErr)
+				level.Debug(util.Logger).Log("Dropped request details: \n%v", requests)
+				continue
```

Comment:
Probably worth pulling these changes out into a separate PR; for instance, we should export a counter of dropped writes at very least, and potentially even drop the entire chunk write somehow.

Reply:
Adding a counter is certainly something I can add in a separate PR. As for dropping the chunk write, currently it drops the entire batch, which can be a bit heavy handed but it didn't seem at the time that the error data from dynamo provided any information to determine what element of the batch was the problem.